packages/pyright-internal/src/analyzer/typePrinter.ts

```diff
@@ -76,7 +79,12 @@ export function printType(
 ): string {
     const parenthesizeUnion = (printTypeFlags & PrintTypeFlags.ParenthesizeUnion) !== 0;
     const parenthesizeCallable = (printTypeFlags & PrintTypeFlags.ParenthesizeCallable) !== 0;
-    printTypeFlags &= ~(PrintTypeFlags.ParenthesizeUnion | PrintTypeFlags.ParenthesizeCallable);
+    const annotationFormat = (printTypeFlags & PrintTypeFlags.AnnotationFormat) !== 0;
+    printTypeFlags &= ~(
+        PrintTypeFlags.ParenthesizeUnion |
+        PrintTypeFlags.ParenthesizeCallable |
+        PrintTypeFlags.AnnotationFormat
```

Comment:
I don't think you want to clear the AnnotationFormat flag here. It should be included recursively for other types that are printed within the outer type.